src/libutil/signature/local-keys.cc

```diff
@@ -14,17 +14,25 @@ BorrowedCryptoValue BorrowedCryptoValue::parse(std::string_view s)
     return {s.substr(0, colon), s.substr(colon + 1)};
 }
 
-Key::Key(std::string_view s)
+Key::Key(std::string_view s, bool sensitiveValue)
 {
     auto ss = BorrowedCryptoValue::parse(s);
 
     name = ss.name;
     key = ss.payload;
 
-    if (name == "" || key == "")
-        throw Error("key is corrupt");
-
-    key = base64Decode(key);
+    try {
+        if (name == "" || key == "")
+            throw FormatError("key is corrupt");
+
+        key = base64Decode(key);
+    } catch (Error & e) {
+        std::string extra;
+        if (!sensitiveValue)
+            extra = fmt(" with raw value '%s'", key);
```

Comment:
Do we still need `sensitiveValue`? I think the context gives enough info.

Reply:
Eh it's pretty self-contained here like the other version (The `Key` constructor is now protected so it won't get used elsewhere).

I am inclined to keep it also because the thing I need to debug is a public key not private key.